File: escolhe_menu.rb
/// <reference types="cypress" />

def choose_menu(menu_path)
  items = menu_path.strip.split(/\s*>\s*/)
  links = "$('#menuDropDown').find('a:contains(\"#{items.shift}\")')"
  items.each do |item|
    links = links + ".parent().find('a:contains(\"#{item}\")').filter(function() {return $(this).text() === \"#{item}\"})"
  end
  page.execute_script(links + "[0].click()")
end



if ARGV[0] == '--start'
    escolhe_menu
end

File: escolhe_menu.js
/// <reference types="cypress" />

function escolhe_menu(menuPath) {
    const items = menuPath.split(' > ');
    let links = '';
    items.forEach((item, index) => {
        links += index === 0
            ? `$('#menuDropDown').find('a:contains("${item}")')`
            : `.parent().find('a:contains("${item}")')`
    },'');
    links + "[0].click()"
}

export { escolhe_menu } 